examples/platforms/simulation/logging.c

```diff
@@ -44,21 +45,81 @@
 #include "utils/code_utils.h"
 
 #if (OPENTHREAD_CONFIG_LOG_OUTPUT == OPENTHREAD_CONFIG_LOG_OUTPUT_PLATFORM_DEFINED)
-OT_TOOL_WEAK void otPlatLog(otLogLevel aLogLevel, otLogRegion aLogRegion, const char *aFormat, ...)
+
+static FILE *sLogFile = NULL;
+
+void platformLoggingSetFileName(const char *aName)
+{
+    if (sLogFile != NULL)
+    {
+        fclose(sLogFile);
+    }
+
+    sLogFile = fopen(aName, "wt");
+
+    if (sLogFile == NULL)
+    {
+        fprintf(stderr, "Failed to open log file '%s', errno %d\r\n", aName, errno);
```

Comment:
is it better to print error string?
```suggestion
        fprintf(stderr, "Failed to open log file '%s': %s\r\n", aName, strerror(errno));
```